### ISSUE TITLE
Crashs if no 'date' key in data.

### ISSUE THREAD

**Issue by odscjames** [open-contracting/ocds-merge#21]
https://github.com/open-contracting/ocds-merge/blob/master/ocdsmerge/merge.py#L316

We are currently seeing crashes in Kingfisher process where there is data with no 'date' key.

In fairness, I'm not sure what the merge library can actually do in this case 😄 ... but maybe we want to think about good ways to report this upwards.

**Comment by pindec** [comment]
Could the issue be that the failing source has _records_ with null release_date fields?

Looks like collection id = 47 compiled OK into a compiled_release collection (id = 48) despite having null release_date fields.

**Comment by odscjames** [comment]
Can I just double check; does it have "null" or does it have "no 'date' key" - these are different.

But actually, a merge operation when the date field is null is probably nonsense! We should probably flag that somehow.

**Comment by pindec** [comment]
Ah, good question. To clarify, my query was on the views.release_summary_with_data table using
`release_date is null `
which I think is effectively "no 'date' key" rather than date = "null".

**Comment by jpmckinney** [comment]
If the date were null, we'd instead get:

```
> sorted([None, '2010-01-01'])
TypeError: '<' not supported between instances of 'str' and 'NoneType'
```

In either case (null value or missing key), merging is undefined, unless the release list contains exactly one release. I implemented more specific exceptions in #22
